azure-core.gemspec

```diff
@@ -26,17 +26,12 @@ Gem::Specification.new do |s|
   s.homepage    = 'http://github.com/Azure/azure-ruby-asm-core'
   s.license     = 'Apache License, Version 2.0'
   s.files       = `git ls-files`.split("\n").reject { |f|  f.start_with?("test/unit") }
+  s.extensions  = 'ext/mkrf_conf.rb'
 
   s.required_ruby_version = '>= 1.9.3'
 
   s.add_runtime_dependency('faraday',                 '~> 0.9')
   s.add_runtime_dependency('faraday_middleware',      '~> 0.10')
-  if RUBY_VERSION < "2.1.0"
```

Comment:
How about adding 'nokogiri' as a development dependency with the condition?
I haven't got a chance to try yet, I assume the followings are true:
1. 'nokogiri' will not be bound when building the gem.
2. 'nokogiri' will be installed when `gem install` or 'bundle install' from the source code of this repo.
3. It won't impact the other projects which have 'azure-core' as rumtime dependency, because the extension script will take care the 'nokogiri' installation. (It seems this can be validated only when new 'azure-core' is released.)